controllers/provisioning_controller.go

```diff
@@ -39,10 +56,29 @@ type ProvisioningReconciler struct {
 // Reconcile updates the cluster settings when the Provisioning
 // resource changes
 func (r *ProvisioningReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
-	_ = context.Background()
-	_ = r.Log.WithValues("provisioning", req.NamespacedName)
+	ctx := context.Background()
+	log := r.Log.WithValues("provisioning", req.NamespacedName)
+
+	infra := &osconfigv1.Infrastructure{}
+	err := r.Client.Get(ctx, client.ObjectKey{
+		Name: "cluster",
+	}, infra)
+	if err != nil {
+		log.Info("Unable to determine Platform that the Operator is running on.")
+		return ctrl.Result{}, err
+	}
 
-	// your logic here
+	// Disable ourselves on platforms other than bare metal
+	if infra.Status.Platform != osconfigv1.BareMetalPlatformType {
```

Comment:
This behavior could be tested with a unit test. Let's add a basic test. I know kubebuilder is recommending ginkgo, but we can just use normal test functions and packages like we do in the other metal3 components for now and consider ginkgo later.

Reply:
Yes, please check the latest now.